docs/spec/lightclient/verification.md

```diff
@@ -844,8 +915,139 @@ If on the blockchain the validator set of the block at height
 
 
 
+## Supporting the IBC Relayer
+
+We might need to download and verify headers whose heights are smaller
+than lightStore.LatestVerified.Height. To formalize
+the solution, we start with the following methods for the LightStore
+
+
+```go
+func (ls LightStore) LatestPrevious(height Height) (LightBlock, bool) 
+```
+- Expected postcondition
+  - returns a light block *lb* that satisfies:
+      - *lb* is in lightStore
+      - *lb* is verified and not expired
+	  - *lb.Header.Height < height*
+	  - for all *b* in lightStore s.t. *b* is verified and not expired it
+        holds *lb.Header.Height >= b.Header.Height*
+   - or false in the second argument if
+    the LightStore does not contain such an *lb*.
+ 
+
+```go
+func (ls LightStore) MinVerified() (LightBlock, bool) 
+```
+- Expected postcondition
+  - returns a light block *lb* that satisfies:
+      - *lb* is in lightStore
+      - *lb* is verified **TODO:** perhaps replace by trusted?
+	  - *lb.Header.Height* is minimal in the lightStore
+	  - **TODO:** according to this it might be expired (outside the
+        trusting period). I believe this is safe. Are there reasons we
+        should not do that?
+   - or false in the second argument if
+    the LightStore does not contain such an *lb*.
+ 
+If a height that is smaller than the smallest height in the lightstore
+is required, we check the hashes backwards. This is done with the
+following function:
+ 
+#### **[LCV-FUNC-BACKWARDS.1]**:
+```go
+func Backwards(primary PeerID, lightStore LightStore,
+                    targetHeight Height) (LightStore, Result) {
+  
+  lb,res = lightStore.MinVerified()
+  if res = false {
+    return (lightStore, ResultFailure)
+  }
+
+  latest := lb.Header
+  for i := lb.Header.height - 1; i >= targetHeight; i-- {
+    // here we download height-by-height. We might first download all
+	// headers down to targetHeight and then check them.
+    current := FetchLightBlock(primary,i)
+    if (hash(current) != latest.Header.LastBlockId) {
+      return (lightStore, ResultFailure)
+    }
+	else {
+	  lightStore.Update(current, StateVerified)
+	  // **TODO:** Do we need a new state type for backwards.
+	}
+    latest := current
+  }
+  return (lightStore, ResultSuccess)
+ }
+```
 
+The following function just decided based on the required height which
+method should be used.
 
+#### **[LCV-FUNC-IBCMAIN.1]**:
```

Comment:
Nit but can we ensure the spacing and line format is consistent here? Should adhere to `go fmt` for readability.

Reply:
Yeah. was a bit ugly. I tried to fix that by hand now. `gofmt` complained as it might be too much pseudo code. But it looks better now. please approve!

Reply:
`gofmt` apparently has a different opinion on documenting things :-)